src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -305,6 +306,42 @@ public function orWhere($column, $operator = null, $value = null)
         return $this->where($column, $operator, $value, 'or');
     }
 
+    /**
+     * Add a "BelongsTo" relationship where clause to the query.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $related
+     * @param  string  $relationship
+     * @param  string  $boolean
+     * @return $this
+     *
+     * @throws \Exception
+     */
+    public function whereBelongsTo($related, $relationshipName = null, $boolean = 'and')
+    {
+        if ($relationshipName === null) {
+            $relationshipName = Str::camel(class_basename(get_class($related)));
+        }
+
+        try {
+            $relationship = $this->model->{$relationshipName}();
+        } catch (BadMethodCallException $exception) {
+            throw RelationNotFoundException::make($this->model, $relationshipName);
+        }
+
+        if (! $relationship instanceof BelongsTo) {
+            throw RelationNotFoundException::make($this->model, $relationshipName, BelongsTo::class);
+        }
+
+        $this->where(
+            $relationship->getForeignKeyName(),
```

Comment:
Do you think `getQualifiedForeignKeyName` should be used here instead?

Reply:
Ah - complex joins. I've made this change.